### PR TITLE
PCA Optimization for Column-Major Tables

### PR DESCRIPTION
**Pull request by owerbat**
Some optimizations for PCA have been done in Correlation

### CODE REVIEW

**Comment by owerbat** [comment]
make test

File: algorithms/kernel/covariance/covariance_impl.i [hunk 1]
```diff
@@ -276,8 +276,11 @@ services::Status updateDenseCrossProductAndSums(bool isNormalized, size_t nFeatu
         default: break;
         }
 
+        DEFINE_TABLE_BLOCK(ReadRows, dataBlock, dataTable);
```
Comment (SmirnovEgorRu) [inline algorithms/kernel/covariance/covariance_impl.i]: So, here we are still using sequential conversion, right?

File: algorithms/kernel/covariance/covariance_impl.i [hunk 2]
```diff
@@ -184,29 +185,28 @@ services::Status updateDenseCrossProductAndSums(bool isNormalized, size_t nFeatu
             algorithmFPType alpha = 1.0;
             algorithmFPType beta  = 1.0;
 
-            size_t startRow = iBlock * numRowsInBlock;
-            size_t endRow   = startRow + numRowsInBlock;
-            if (endRow > nVectors)
-            {
-                endRow = nVectors;
-            }
+            size_t _nRows    = (iBlock < (numBlocks - 1)) ? numRowsInBlock : numRowsInLastBlock;
```
Comment (SmirnovEgorRu) [inline algorithms/kernel/covariance/covariance_impl.i]: naming of variables with _* prefix is used for private fields in classes only

File: algorithms/kernel/covariance/covariance_impl.i (hunk 1 above)
Comment (SmirnovEgorRu) [reply algorithms/kernel/covariance/covariance_impl.i]: Does it affect PCA?

File: algorithms/kernel/covariance/covariance_impl.i (hunk 2 above)
Comment (owerbat) [reply algorithms/kernel/covariance/covariance_impl.i]: fixed

**Pull request merged**